Mathlib/AlgebraicGeometry/EllipticCurve/NormalForms.lean

```diff
@@ -101,10 +100,10 @@ variable {K : Type*} [Field K] (W₁ : WeierstrassCurve K) (E : EllipticCurve K)
 
```

Comment:
Unsure about this: is `Type*` weaker than `Type u` and `Type v` for explicit universes `u` and `v`? I try to make them as general as possible so far. I think I encountered some issues when trying to use `Type*` in the number field files.

For my projective/Jacobian files I tend to stick with `W' : WeierstrassCurve R` for Weierstrass curves over rings (instead of `W₁` and `W : WeierstrassCurve F` for Weierstrass curves over fields (because I wanted to reserve `K` for specific fields like number fields or finite fields), but feel free to ignore this since this is a different file.

Reply:
> Unsure about this: is `Type*` weaker than `Type u` and `Type v` for explicit universes `u` and `v`?

`Type*` is the same as `Type u` and `Type v` as long as you don't need to specify universe levels manually. Each `Type*` will be interpreted as distinct universe levels, for example, `{R K : Type*}` means that `R : Type u_1` and `K : Type u_2`. The issue occurred when you want some of these objects are in the same universe level, or if you want to mention the level manually (for example, Cardinal.lift, max of universe levels, e.g. in [lift_rank_mul_lift_rank](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/Dimension/Free.html#lift_rank_mul_lift_rank)).

> For my projective/Jacobian files I tend to stick with `W' : WeierstrassCurve R` for Weierstrass curves over rings (instead of `W₁` and `W : WeierstrassCurve F` for Weierstrass curves over fields (because I wanted to reserve `K` for specific fields like number fields or finite fields), but feel free to ignore this since this is a different file.

I see. In fact the name `W₁` is for laziness. Ideally I should use `W` for all `WeierstrassCurve`.